### ISSUE TITLE
Hashimoto exploit failing on mt6789 / V6 bootrom

### ISSUE THREAD

**Issue by ryenyuku** [bkerler/mtkclient#826]
Might be a relevant issue with #758, but it seems to be a different problem.

`ryenyuku@Advan-Laptop /m/D/P/I/N/m> python mtk printgpt --ptype hashimoto
MTK Flash/Exploit Client V1.6.3 (c) B.Kerler 2018-2023

Preloader - Status: Waiting for PreLoader VCOM, please connect mobile

Port - Hint:

Power off the phone before connecting.
For brom mode, press and hold vol up, vol dwn, or all hw buttons and connect usb.
For preloader mode, don't press any hw button and connect usb.
If it is already connected and on, hold power for 10 seconds to reset.

...........

Port - Hint:

Power off the phone before connecting.
For brom mode, press and hold vol up, vol dwn, or all hw buttons and connect usb.
For preloader mode, don't press any hw button and connect usb.
If it is already connected and on, hold power for 10 seconds to reset.

...........
Port - Device detected :)
Preloader -     CPU:                    MT6789(MTK Helio G99)
Preloader -     HW version:             0x0
Preloader -     WDT:                    0x10007000
Preloader -     Uart:                   0x11002000
Preloader -     Brom payload addr:      0x100a00
Preloader -     DA payload addr:        0x201000
Preloader -     Var1:                   0xa
Preloader - Disabling Watchdog...
Preloader - HW code:                    0x1208
Preloader - Target config:              0xe5
Preloader -     SBC enabled:            True
Preloader -     SLA enabled:            False
Preloader -     DAA enabled:            True
Preloader -     SWJTAG enabled:         True
Preloader -     EPP_PARAM at 0x600 after EMMC_BOOT/SDMMC_BOOT:  False
Preloader -     Root cert required:     False
Preloader -     Mem read auth:          True
Preloader -     Mem write auth:         True
Preloader -     Cmd 0xC8 blocked:       True
Preloader - Get Target info
Preloader - BROM mode detected.
Preloader -     HW subcode:             0x8a00
Preloader -     HW Ver:                 0xca00
Preloader -     SW Ver:                 0x0
Preloader - ME_ID:                      8E7E332B7EA4A1E04C788B69716F5E73
PLTools - Loading payload from generic_patcher_payload.bin, 0x538 bytes
Exploitation - Hashimoto Run
hwcrypto - Disable Caches
hwcrypto - CQDMA Disable Range Blacklist
cqdma - Disabling bootrom range checks..
Exploitation - Sending payload
Exploitation
Exploitation - [LIB]: ←[31mFailed to load payload file. Error: required argument is not an integer←[0m
PLTools - Successfully sent payload: /mnt/Data/Projects/Infinix/Note 30/mtkclient/mtkclient/payloads/generic_patcher_payload.bin
DA_handler - Device is protected.
DA_handler - Device is in BROM mode. Trying to dump preloader.
Traceback (most recent call last):
  File "/mnt/Data/Projects/Infinix/Note 30/mtkclient/mtk", line 855, in <module>
    mtk = Main(args).run(parser)
          ^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/Data/Projects/Infinix/Note 30/mtkclient/mtkclient/Library/mtk_main.py", line 635, in run
    mtk = da_handler.configure_da(mtk, preloader)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/Data/Projects/Infinix/Note 30/mtkclient/mtkclient/Library/DA/mtk_da_handler.py", line 119, in configure_da
    if not mtk.daloader.upload_da(preloader=preloader):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/Data/Projects/Infinix/Note 30/mtkclient/mtkclient/Library/DA/mtk_daloader.py", line 239, in upload_da
    self.set_da()
  File "/mnt/Data/Projects/Infinix/Note 30/mtkclient/mtkclient/Library/DA/mtk_daloader.py", line 182, in set_da
    self.da = DAXML(self.mtk,self.daconfig,self.loglevel)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/Data/Projects/Infinix/Note 30/mtkclient/mtkclient/Library/DA/xml/xml_lib.py", line 63, in __init__
    self.partition = Partition(self.mtk, self.readflash, self.read_pmt, loglevel)
                                         ^^^^^^^^^^^^^^
AttributeError: 'DAXML' object has no attribute 'readflash'`

**Comment by ryenyuku** [comment]
> get preloader from device firmware try again by using --preloader option
> 
> although it won't work but for better logs

Alright, I'll do it soon. I just recovered this thing from graveyard using a paid tool.

**Comment by Shakib-BD** [comment]
> > get preloader from device firmware try again by using --preloader option
> > although it won't work but for better logs
> 
> Alright, I'll do it soon. I just recovered this thing from graveyard using a paid tool.

Which device is it?

**Comment by ryenyuku** [comment]
> Which device is it?

Infinix Note 30 (X6883B), fixed it using Android Multi Tool with 10€ wasted

**Comment by seedmonn** [comment]
> > Which device is it?
> 
> Infinix Note 30 (X6883B), fixed it using Android Multi Tool with 10€ wasted

Can you contact me please on telegram @seedmonn.
Is that [you](https://t.me/ryenyuku)?

**Comment by ryenyuku** [comment]
> Can you contact me please on [telegram](https://t.me/seedmon). Is that [you](https://t.me/ryenyuku)?

Yes, but why do you want to contact me privately? Plus I'm lazy to login to my Telegram account, just contact me via Discord (@ryenyuku)

**Comment by seedmonn** [comment]
> > Can you contact me please on [telegram](https://t.me/seedmon). Is that [you](https://t.me/ryenyuku)?
> 
> Yes, but why do you want to contact me privately? Plus I'm lazy to login to my Telegram account, just contact me via Discord (@ryenyuku)

I was need your advice w/ unbricking INFINIX Note 30, but i managed
Sorry for disturbing

**Comment by bkerler** [comment]
This has nothing to do with hashimoto exploit and double of #758.
